backend/src/test/java/ai/verta/modeldb/TestsInit.java

```diff
@@ -73,8 +73,13 @@ public class TestsInit {
   protected static HydratedServiceGrpc.HydratedServiceBlockingStub
       hydratedServiceBlockingStubClient2;
   protected static ArtifactStoreGrpc.ArtifactStoreBlockingStub artifactStoreBlockingStub;
+  private static Boolean setServerAndServiceRan = false;
```

Comment:
I think this should be an AtomicBoolean with proper CAS access to make sure there aren't race conditions with the initialization.

Reply:
We are not executing test classes parallelly so race condition will never accrued. but lets make it for safety.

Reply:
Fixed it.

Reply:
The other reason would be cross-thread visibility. I don't think there's any guarantee that junit will run all the tests on the same thread, even if they aren't running concurrently. Thanks for the change! 👍